protocol/directory_test.go

```diff
@@ -138,3 +139,231 @@ func TestRegisterExistedUserWithoutTB(t *testing.T) {
 		t.Fatal("Expect returned TB is nil")
 	}
 }
+
+func TestKeyLookupWithTB(t *testing.T) {
+	d := newDirectory(t, true)
+	_, err := d.Register(&RegistrationRequest{
```

Comment:
I think it might be worth keeping the response message here to make sure the server responds with the same TB for the following lookup, as a sanity check.